### ISSUE TITLE
Possible API improvement

### ISSUE THREAD

**Issue by ryanto** [simplabs/ember-test-selectors#18]
The current testSelector API feels a little repetitive since it repeats the word selector twice.

```
find(testSelector('selector', 'post-title')) // => find('[data-test-selector="post-title"]')
```

It would be nice if `testSelector('post-title')` would return `[data-test-selector="post-title"]`, but I know that would introduce a breaking change. I'm wondering if there's another more concise helper we can use... maybe one of the following would be suitable?

```
testElement('post-title') // => [data-test-selector="post-title"]
testData('post-title') // => [data-test-selector="post-title"]
testAttribute('post-title') // => [data-test-selector="post-title"]
```

Let me know what you think and if you like any of the above. I'll PR if you're ok with it.

PS Thanks for the amazing addon!

**Comment by marcoow** [comment]
We are at version 0.0.3 to breaking changes aren't an issue fortunately ;)

The point is that we need to keep the attribute suffix configurable. We could maybe use sth. like

```js
find(testSelector('post-title')) // => find('*[data-test-selector="post-title"]')
find(testSelector(6, 'post-id')) // => find('*[data-test-post-id="post-title"]')
```

/cc @pangratz

**Comment by ryanto** [comment]
If breaking changes are ok how do you feel about `testSelector` and `testSelectorById`? I think explicit function names might be easier to follow.

**Comment by marcoow** [comment]
how would those work exactly?

**Comment by ryanto** [comment]
```javascript
testSelector('post-title'); // => '[data-test-selector="post-title"]'
testSelectorById('post-title', 6); // => '[data-test-selector-post-title="6"]'
```

**Comment by marcoow** [comment]
Hm, I don't really like the idea of having 2 separate methods. Basically what we want in order to generate these selectors is:

* `[data-test-selector="post-title"]`: attribute value `post-title` must be specifiable of course, attribute suffix (`-selector` in this case) should be configurable while `-selector` should probably be the default
* `[data-test-post-title="6"]`: attribute value `6` gets passed as an argument to `testSelector` of course, attribute suffix (`-post-title` in this case) should get passed as well, overriding the default argument value of `-selector`

so I think my above proposal would work well for both of these cases?

```js
find(testSelector('post-title')) // => find('*[data-test-selector="post-title"]')
find(testSelector(6, 'post-id')) // => find('*[data-test-post-id="6"]')
```

To make that clearer, declaration of `testSelector` would be

```js
testSelector(attributeValue, attributeSuffix = 'selector') {
  …
```

**Comment by ryanto** [comment]
Ah makes sense.

I'll PR this sometime in the next few days.

**Comment by pdud** [comment]
Heyo!

I believe you can avoid this repetition without making any changes to the code base. What do you think of the following?

I've been using test selectors like  `<h1 data-test-post-title>` and then selecting it with `testSelector('post-title')`. This is even less repetition as `selector` is not needed in the javascript or html.

Granted you can't easily make it dynamic but in those instances I've found `<h1 data-test-post-id="6">` to be a better fit anyhow.

**Comment by marcoow** [comment]
@pdud: you mean `testSelector('post-title')` would result in `find('*[data-test-selector="post-title"]')`? I'm not sure it's so great combining construction of the selector and using it in `find` as you might also want to use it for `click`, `fillIn` etc.

**Comment by pdud** [comment]
Oops I wasn't clear enough.

I'm questioning the need for the proposed change in favor of doing something like this which already works with the library as is.

`find(testSelector('post-title'))` would result in `find('[data-test-post-title]')` and match the template `<h1 data-test-post-title>My First Post</h1>`

No need to add the suffix `-selector` anywhere by simply not using a value on your data-attribute in the html or javascript.

So basically in your html do `<h1 data-test-post-title>My First Post</h1>` rather than `<h1 data-test-selector="post-title">My First Post</h1>`

**Comment by marcoow** [comment]
Ah I see, that makes sense. So you're suggesting

```js
find(testSelector('post-title')) // => find('*[data-test-post-title]')
find(testSelector('post-id', 6)) // => find('*[data-test-post-id="6"]')
```

right?

**Comment by pdud** [comment]
Yeap! Using it as you specified in your comment directly above has worked well for me :)

**Comment by marcoow** [comment]
sounds good - what do you think @ryanto?

**Comment by ryanto** [comment]
Oh nice find @pdud. This works perfectly. I'll close this issue.

Thanks for the feedback and advice @marcoow
